# Patch to add support for Python 3.9, 3.10, 3.11
# Combined patches for 3.9 & 3.10 into a single patch which included 3.11
# The previous patches were no longer accurate anyway
# CW Gum <a108384@gmail.com>
# Aug 24, 2024
#

diff --git a/configure.ac b/configure.ac
index 5b432eae62..1691c84561 100644
--- a/configure.ac
+++ b/configure.ac
@@ -93,17 +93,18 @@ SR_PKG_CHECK_SUMMARY([srd_pkglibs_summary])
 # first, since usually only that variant will add "-lpython3.8".
 # https://docs.python.org/3/whatsnew/3.8.html#debug-build-uses-the-same-abi-as-release-build
 SR_PKG_CHECK([python3], [SRD_PKGLIBS],
-	[python-3.8-embed], [python-3.8 >= 3.8], [python-3.7 >= 3.7], [python-3.6 >= 3.6], [python-3.5 >= 3.5], [python-3.4 >= 3.4], [python-3.3 >= 3.3], [python-3.2 >= 3.2], [python3 >= 3.2])
+	[python-3.12-embed], [python-3.11-embed], [python-3.10-embed], [python-3.9-embed], [python-3.8-embed], [python3.11 >= 3.11], [python3.10 >= 3.10], [python-3.9 >= 3.9], [python-3.8 >= 3.8], [python-3.7 >= 3.7], [python-3.6 >= 3.6], [python-3.5 >= 3.5], [python-3.4 >= 3.4], [python-3.3 >= 3.3], [python-3.2 >= 3.2], [python3 >= 3.2])
 AS_IF([test "x$sr_have_python3" = xno],
 	[AC_MSG_ERROR([Cannot find Python 3 development headers.])])
 
 # We also need to find the name of the python3 executable (for 'make install').
 # Some OSes call this python3, some call it python3.2, etc. etc.
 AC_ARG_VAR([PYTHON3], [Python 3 interpreter])
-AC_CHECK_PROGS([PYTHON3], [python3.8 python3.7 python3.6 python3.5 python3.4 python3.3 python3.2 python3])
+AC_CHECK_PROGS([PYTHON3], [python3.12, python3.11, python3.10, python3.9, python3.8 python3.7 python3.6 python3.5 python3.4 python3.3 python3.2 python3])
 AS_IF([test "x$PYTHON3" = x],
 	[AC_MSG_ERROR([Cannot find Python 3 interpreter.])])
 
+
 ######################
 ##  Feature checks  ##
 ######################
